src/main/java/com/files/ValidateImages.java

```diff
@@ -0,0 +1,42 @@
+package com.files;
+
+import com.exeptions.ExtensionNotSupported;
+import com.image.Extension;
+import com.image.Image;
+import com.image.Images;
+import com.log.Logs;
+
+import java.util.Iterator;
+import java.util.logging.Logger;
+
+public class ValidateImages implements Images {
+
+    Images images;
+
+    public ValidateImages(Images images) {
+        this.images = images;
+    }
+
+    @Override
+    public void addImage(Image image) {
```

Comment:
@PavelNejinskiy please avoid compound names like this. Current scope - implementation of `Images` interface, there is no need to have it like `addImage`, we can use just `add` since we are adding to images instance right?

So you the client code can use it like
```
Images images = new ValdateImages(...);
images.add(..);
```
vs `images.addImage(..)`

This applies for other methods also, like `Image.getExtension`. It is much more readable if we have it like `Image.extension` we do not need get part. Please read
http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html
and also
http://www.yegor256.com/2014/09/16/getters-and-setters-are-evil.html